cmd/operator-sdk/main.go

```diff
@@ -15,32 +15,51 @@
 package main
 
 import (
-	"os"
 
 	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
 	// to ensure that `exec-entrypoint` and `run` can make use of them.
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 
 	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/cli"
 	kbutil "github.com/operator-framework/operator-sdk/internal/util/kubebuilder"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
 
 	log "github.com/sirupsen/logrus"
 )
 
 func main() {
-	// Use the new KB CLI only when running inside an existing Kubebuilder project with a PROJECT file.
-	// The default legacy CLI provides the init cmd to initialize
-	// a Kubebuilder project as a way to opt into the new KB CLI.
-	// TODO: Make the new KB CLI the default, once the integration is complete
-	// and deprecate "operator-sdk new" from the old CLI.
+	// Use the new KB CLI when running inside a Kubebuilder project with an existing PROJECT file.
 	if kbutil.HasProjectFile() {
 		if err := cli.Run(); err != nil {
 			log.Fatal(err)
 		}
 		return
 	}
 
-	if err := cli.RunLegacy(); err != nil {
-		os.Exit(1)
+	// Use the legacy CLI if inside of a Go/Helm/Ansible legacy project
+	operatorType := projutil.GetOperatorType()
+	switch operatorType {
+	case projutil.OperatorTypeGo, projutil.OperatorTypeHelm, projutil.OperatorTypeAnsible:
+		// Deprecation warning for Go projects
+		// TODO/Discuss: UX wise, is displaying this notice on every command that runs
+		// in the legacy Go projects too loud.
+		if operatorType == projutil.OperatorTypeGo {
+			log.Warn("DEPRECATION NOTICE:\n" +
```

Comment:
It would be better to adhere to [this](https://github.com/operator-framework/operator-sdk/blob/master/internal/util/projutil/project_util.go#L324) deprecation notice formatting.

Reply:
I'm always wary of mixing of logrus and printfs but that worked out.
![image](https://user-images.githubusercontent.com/6660644/84216558-c3744680-aa7e-11ea-81f1-ca1d90760f95.png)

Curiously, it doesn't seem like anything else was using `projutil.PrintDeprecationWarning()` so I've repurposed it for my use here.

Reply:
This is messing with the output of `operator-sdk print-deps` in one of the test-framework sanity check scripts causing it to fail.
https://github.com/operator-framework/operator-sdk/blob/e68fff6b33c46b1156e5648407a291174b519da0/hack/generate/test-framework/gen-test-framework.sh#L15

I think it was okay with logrus instead of `fmt.Printf()` so might have to revert this.

Reply:
Print to stderr? At this point either that or logging works since no other commands use that deprecation notice

Reply:
Logrus was writing to stderr [by default](https://github.com/sirupsen/logrus/blob/master/logger.go#L19) so that works. 
Changing it to stderr.